src/components/application_manager/rpc_plugins/app_service_rpc_plugin/src/app_service_rpc_plugin.cc

```diff
@@ -79,6 +79,9 @@ void AppServiceRpcPlugin::OnApplicationEvent(
   if (plugins::ApplicationEvent::kApplicationRegistered == event) {
     application->AddExtension(
         std::make_shared<AppServiceAppExtension>(*this, *application));
+  } else if (plugins::ApplicationEvent::kApplicationUnregistered == event) {
+    application->RemoveExtension(
```

Comment:
@OlhaVorobiova I think we also need to call `DeleteSubscriptions` when app is unregistered